### ISSUE TITLE
Spike: Investigate if NaturalLanguage should be a parent of some Logical Type

### ISSUE THREAD

**Issue by gsheni** [alteryx/woodwork#742]
- For our type system, we make it so that NaturalLanguage is the parent of some Logical Types. See [here](https://github.com/alteryx/woodwork/blob/main/woodwork/type_sys/type_system.py#L58)
- However, does this actually make sense? NaturalLanguage should be for long-form strings, and things which cannot be inferred as other Logical Types. 
- We should discuss this behavior to make sure it is proper

**Comment by kmax12** [comment]
>NaturalLanguage should be for long-form strings 

i think natural language just means free-form text, not sure why length matters. 

> things which cannot be inferred as other Logical Types.

isn't that the whole point of have the parent relationship. someone's name (e.g "Max Kanter") is both `NaturalLanguage` and `PersonFullName`, `PersonFullName` is just much specific.

**Comment by gsheni** [comment]
Closing this issue as I misinterpreted out parent-child relationship in Woodwork. We have a parent-child relationship in terms of inference. If a Logical Type is not interpreted as a specific Logical Type, it defaults to Natural Language. That is the code [here](https://github.com/alteryx/woodwork/blob/main/woodwork/type_sys/type_system.py#L58).

I agree that NaturalLanguage can be free-form text, regardless of length.
